In [1]:
import sys
import time
import traceback

import models.mnist_data as mnist
from models.cnn_model1 import LeNet5
from modules.hpvconf import HPVGenerator
from modules.hpmgr import HPVManager
from modules.trainmgr import TrainingManager

In [3]:
LOG_PATH = "../log/learning_rate_test1.csv"
TEMPLATE_PATH = 'CNN_HPV.ini'
CONFIG_PATH = '.config/'
NUM_GPUS = 1 # only one gpu can be used for HPOlib

In [5]:
def run(params):
    try:        
        generator = HPVGenerator()
        generator.setTemplate('CNN_HPV.ini')
        hpv_file = generator.generate(params, output_dir=CONFIG_PATH)
                
        train_manager = TrainingManager(LeNet5(mnist.import_dataset()), LOG_PATH)
        train_manager.setTrainingDevices('gpu', NUM_GPUS)
        logging_params = [param for param in params]
        train_manager.setLoggingParams(logging_params)
        hpv = HPVManager(hpv_file, ini_dir=CONFIG_PATH)
        return train_manager.run(hpv)
           
    except:
        e = sys.exc_info()
        print("Error: " + str(e))
        traceback.print_exc()    

In [4]:
use_decays = [True, False]
learning_rates = [0.1, 0.01, 0.0005]
learning_rate_results = {}
for use_decay in use_decays:
    for learning_rate in learning_rates:
        params = {'USE_LEARNING_RATE_DECAY': use_decay, 'BASE_LEARNING_RATE': learning_rate}
        result = run(params)
        #print str(params) + " -> " + str(result)
        learning_rate_results[str(params)] = result


Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14793166815787.ini is terminated properly


TypeError: unhashable type: 'dict'

In [5]:
use_decays = [True, False]
learning_rates = [0.005, 0.02, 0.05, 0.001, 0.0001]

for use_decay in use_decays:
    for learning_rate in learning_rates:
        params = {'USE_LEARNING_RATE_DECAY': use_decay, 'BASE_LEARNING_RATE': learning_rate}
        result = run(params)
        learning_rate_results[str(params)] = result
#print learning_rate_results

Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786315748617.ini is terminated properly
{'BASE_LEARNING_RATE': 0.005, 'USE_LEARNING_RATE_DECAY': True} -> 0.82
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786316795831.ini is terminated properly
{'BASE_LEARNING_RATE': 0.02, 'USE_LEARNING_RATE_DECAY': True} -> 0.6
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786317871794.ini is terminated properly
{'BASE_LEARNING_RATE': 0.05, 'USE_LEARNING_RATE_DECAY': True} -> 0.67
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786318938611.ini is terminated properly
{'BASE_LEARNING_RATE': 0.001, 'USE_LEARNING_RATE_DECAY': True} -> 1.23
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786320018503.ini is terminated properly
{'BASE_LEARNING_RATE': 0.0001, 'USE_LEARNING_RATE_DECAY': True} -> 3.56
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786321104892.ini is terminated properly
{'BASE_LEARNING_RATE': 0.005, 'USE_LEARNING_RATE_DECAY': False} -> 0.79
Training wi

* With learning rate decay
 * **{'BASE_LEARNING_RATE': 0.1}** -> 1.02
 * {'BASE_LEARNING_RATE': 0.05} -> 0.67
 * {'BASE_LEARNING_RATE': 0.02} -> 0.6
 * **{'BASE_LEARNING_RATE': 0.01}** -> 0.63
 * {'BASE_LEARNING_RATE': 0.005} -> 0.82
 * {'BASE_LEARNING_RATE': 0.001} -> 1.23
 * {'BASE_LEARNING_RATE': 0.0005} -> 1.51
 * **{'BASE_LEARNING_RATE': 0.0001}** -> 3.56
 

* Without learning rate decay
 * **{'BASE_LEARNING_RATE': 0.1}** -> 1.4
 * {'BASE_LEARNING_RATE': 0.05} -> 0.85
 * {'BASE_LEARNING_RATE': 0.02} -> 0.77
 * **{'BASE_LEARNING_RATE': 0.01}** -> 0.74
 * {'BASE_LEARNING_RATE': 0.005} -> 0.79
 * {'BASE_LEARNING_RATE': 0.001} -> 1.14
 * {'BASE_LEARNING_RATE': 0.0005} -> 1.29
 * **{'BASE_LEARNING_RATE': 0.0001}** -> 2.91
 

- tested learning rate: 0.1, 0.05, 0.02, 0.01, 0.05, 0.001, 0.0005, 0.0001 (w/ and w/o learning rate decay (0.95))
  * test error decreased near 0.02~0.01 and increased again 
    * test errors w/ learning late decay : 1.02, 0.67, 0.6, 0.63, 0.82,  1.23, 1.51, 3.56
    * test errors w/o  learning late decay : 1.4, 0.85, 0.77, 0.74, 0.79, 1.14, 1.29, 2.91

In [6]:
LOG_PATH = "../log/regularization_test.csv"

In [8]:
regularization_factors = [0, 5e-5, 1e-4, 2e-4, 3e-4, 4e-4, 6e-4, 7e-4, 8e-4, 9e-4, 5e-3, 5e-2]
regularization_results = {}
for regularization_factor in regularization_factors:
    params = {'REGULARIZER_FACTOR': regularization_factor}
    result = run(params)
    #print str(params) + " -> " + str(result) + "%"
    regularization_results[str(params)] = result

Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786746824296.ini is terminated properly
{'REGULARIZER_FACTOR': 0} -> 0.6%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786747884366.ini is terminated properly
{'REGULARIZER_FACTOR': 5e-05} -> 0.59%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786748958402.ini is terminated properly
{'REGULARIZER_FACTOR': 0.0001} -> 0.64%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786750034210.ini is terminated properly
{'REGULARIZER_FACTOR': 0.0002} -> 0.63%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786751109682.ini is terminated properly
{'REGULARIZER_FACTOR': 0.0003} -> 0.69%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786752184817.ini is terminated properly
{'REGULARIZER_FACTOR': 0.0004} -> 0.67%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786753260488.ini is terminated properly
{'REGULARIZER_FACTOR': 0.0006} -> 0.67%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786754353644.in

In [9]:
regularization_factors = [5e-8, 5e-7, 5e-6, 5e-1, 4e-1, 3e-1, 2e-1, 1e-1]

for regularization_factor in regularization_factors:
    params = {'REGULARIZER_FACTOR': regularization_factor}
    result = run(params)
    #print str(params) + " -> " + str(result) + "%"
    regularization_results[str(params)] = result

Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786787480803.ini is terminated properly
{'REGULARIZER_FACTOR': 5e-08} -> 0.67%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786788520223.ini is terminated properly
{'REGULARIZER_FACTOR': 5e-07} -> 0.67%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786789605891.ini is terminated properly
{'REGULARIZER_FACTOR': 5e-06} -> 0.7%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786790696790.ini is terminated properly
{'REGULARIZER_FACTOR': 0.5} -> 4.32%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786791791477.ini is terminated properly
{'REGULARIZER_FACTOR': 0.4} -> 3.16%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786792882536.ini is terminated properly
{'REGULARIZER_FACTOR': 0.3} -> 2.97%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786793987900.ini is terminated properly
{'REGULARIZER_FACTOR': 0.2} -> 2.64%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786795079631.ini is term

* {'REGULARIZER_FACTOR': 0} -> 0.6%
* {'REGULARIZER_FACTOR': 5e-08} -> 0.67%
* {'REGULARIZER_FACTOR': 5e-07} -> 0.67%
* {'REGULARIZER_FACTOR': 5e-06} -> 0.7%
* {'REGULARIZER_FACTOR': 5e-05} -> 0.59%
* {'REGULARIZER_FACTOR': 0.0001} -> 0.64%
* {'REGULARIZER_FACTOR': 0.0002} -> 0.63%
* {'REGULARIZER_FACTOR': 0.0003} -> 0.69%
* {'REGULARIZER_FACTOR': 0.0004} -> 0.67%
* {'REGULARIZER_FACTOR': 0.0006} -> 0.67%
* {'REGULARIZER_FACTOR': 0.0007} -> 0.73%
* {'REGULARIZER_FACTOR': 0.0008} -> 0.68%
* {'REGULARIZER_FACTOR': 0.005} -> 0.99%
* {'REGULARIZER_FACTOR': 0.05} -> 2.37%
* {'REGULARIZER_FACTOR': 0.1} -> 2.12%
* {'REGULARIZER_FACTOR': 0.2} -> 2.64%
* {'REGULARIZER_FACTOR': 0.3} -> 2.97%
* {'REGULARIZER_FACTOR': 0.4} -> 3.16%
* {'REGULARIZER_FACTOR': 0.5} -> 4.32%

After regularizer factor testing, followings may be effective:
 * 0 ~ 0.001 : 0.6? % -> choose **5e-05**
 * 0.005 ~ 0.05 : 0.99% ~ 2.37% -> choose **0.02**
 * other : choose **0.5**  

In [10]:
LOG_PATH = "../log/dropout_test.csv"

In [13]:
dropouts = [1, 0.9, 0.8, 0.75, 0.7, 0.6, 0.4, 0.3, 0.25, 0.2, 0.1]
dropouts_result = {}
for dropout in dropouts:
    params = {'DROPOUT_RATE': dropout}
    result = run(params)
    #print str(params) + " -> " + str(result) + "%"
    dropouts_result(str(params)) = result

Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786818570633.ini is terminated properly
{'DROPOUT_RATE': 0.9} -> 0.77%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786819626294.ini is terminated properly
{'DROPOUT_RATE': 0.8} -> 0.77%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786820733148.ini is terminated properly
{'DROPOUT_RATE': 0.75} -> 0.73%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786821833540.ini is terminated properly
{'DROPOUT_RATE': 0.7} -> 0.69%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786822936462.ini is terminated properly
{'DROPOUT_RATE': 0.6} -> 0.75%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786824038998.ini is terminated properly
{'DROPOUT_RATE': 0.4} -> 0.68%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786825145771.ini is terminated properly
{'DROPOUT_RATE': 0.3} -> 0.75%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786826249595.ini is terminated properly
{'DROPOUT_RATE': 0.25} -> 0.58

Traceback (most recent call last):
  File "models/cnn_model.py", line 79, in learn
    y = self.train()
  File "models/cnn_model.py", line 107, in train
    logits = self.leNet5(self.train_data_node, True)
  File "models/cnn_model.py", line 392, in leNet5
    hidden = tf.nn.dropout(hidden, 1.0 - self.hp["DROPOUT_RATE"], seed=self.hp["SEED"])
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/nn_ops.py", line 1114, in dropout
    "range (0, 1], got %g" % keep_prob)
ValueError: keep_prob must be a scalar tensor or a float in the range (0, 1], got 0


following dropout rates are selected: **1.0, 0.75, 0.25**

In [18]:
LOG_PATH = "../log/reg_mix_test.csv"

In [16]:
learning_rates = [0.1, 0.01, 0.0001]
regularization_factors = [5e-05, 0.025, 0.5]
dropouts = [1., 0.75, 0.25]
mixed_results = []
for use_decay in use_decays:
    for learning_rate in learning_rates:
        for regularization_factor in regularization_factors:
            for dropout in dropouts:    
                params = {'BASE_LEARNING_RATE': learning_rate, \
                      'REGULARIZER_FACTOR': regularization_factor, 'DROPOUT_RATE': dropout}
                result = run(params)
                #print str(params) + " -> " + str(result) + "%"
                mixed_results[str(params)] = result

Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786915277472.ini is terminated properly
{'DROPOUT_RATE': 1.0, 'BASE_LEARNING_RATE': 0.1, 'REGULARIZER_FACTOR': 5e-05, 'USE_LEARNING_RATE_DECAY': True} -> 1.71%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786916317935.ini is terminated properly
{'DROPOUT_RATE': 0.75, 'BASE_LEARNING_RATE': 0.1, 'REGULARIZER_FACTOR': 5e-05, 'USE_LEARNING_RATE_DECAY': True} -> 89.72%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786917411138.ini is terminated properly
{'DROPOUT_RATE': 0.25, 'BASE_LEARNING_RATE': 0.1, 'REGULARIZER_FACTOR': 5e-05, 'USE_LEARNING_RATE_DECAY': True} -> 89.72%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786918492949.ini is terminated properly
{'DROPOUT_RATE': 1.0, 'BASE_LEARNING_RATE': 0.1, 'REGULARIZER_FACTOR': 0.25, 'USE_LEARNING_RATE_DECAY': True} -> 90.2%
Training with /root/tf-hpolib/tf-hpo/src/config/HPV_14786919573739.ini is terminated properly
{'DROPOUT_RATE': 0.75, 'BASE_LEARNING_RATE': 0.

In [3]:
def test_run(params):
    try:        
        generator = HPVGenerator()
        generator.setTemplate(TEMPLATE_PATH)
        hpv_file = generator.generate(params, output_dir=CONFIG_PATH)
                
        train_manager = TrainingManager(LeNet5(mnist.import_dataset()))
        train_manager.setTrainingDevices('gpu', NUM_GPUS)
        
        hpv = HPVManager(hpv_file, ini_dir=CONFIG_PATH)
        return train_manager.run(hpv)
           
    except:
        e = sys.exc_info()
        print("Error: " + str(e))
        traceback.print_exc()    

In [4]:
LOG_PATH = "../log/conv_params_test.csv"

In [6]:
# convolutional network hyperparameters 
paddings = ['VALID']#['SAME', 'VALID']
maxpooling_window_sizes = [4, 6, 8, 10] # [1, 2, 4, 6, 8, 10]
maxpooling_stride_sizes = [2]
conv_results = {}
for padding in paddings:
    for maxpooling_window_size in maxpooling_window_sizes:
        for maxpooling_stride_size in maxpooling_stride_sizes:
            params = {'PADDING':padding, 'POOLING_SIZE': maxpooling_window_size, 'STRIDE_SIZE' : maxpooling_stride_size }
            result = run(params)
            #print str(params) + " -> " + str(result) + "%"
            conv_results[str(params)] = result
for key in conv_results:
    print key + " -> " + str(conv_results[key]) +  "%"

({'conv2_depth': 64, 'dropout_rate': 0.5, 'conv1_depth': 32, 'OPTIMIZATION': 'Momentum', 'MOMENTUM_VALUE': 0.9, 'base_learning_rate': 0.01, 'filter_size': 5, 'init_stddev': 0.1, 'seed': 66478, 'BATCH_SIZE': 64, 'pooling_size': 4, 'num_epochs': 20, 'eval_frequency': 100, 'eval_batch_size': 64, 'FC1_WIDTH': 512, 'fc1_width': 512, 'momentum_value': 0.9, 'STRIDE_SIZE': 2, 'decay_rate': 0.95, 'optimization': 'Momentum', 'num_pooling': 2, 'POOLING_SIZE': 4, 'NUM_EPOCHS': 20, 'USE_LEARNING_RATE_DECAY': True, 'EVAL_FREQUENCY': 100, 'EVAL_BATCH_SIZE': 64, 'regularizer_factor': 0.0005, 'batch_size': 64, 'BASE_LEARNING_RATE': 0.01, 'padding': 'VALID', 'FILTER_SIZE': 5, 'INIT_WEIGHT_VALUE': 0.1, 'use_learning_rate_decay': True, 'CONV2_DEPTH': 64, 'DROPOUT_RATE': 0.5, 'CONV1_DEPTH': 32, 'DECAY_RATE': 0.95, 'REGULARIZER_FACTOR': 0.0005, 'INIT_STDDEV': 0.1, 'stride_size': 2, 'PADDING': 'VALID', 'SEED': 66478, 'NUM_POOLING': 2, 'init_weight_value': 0.1},)
({'VALIDATION': False, 'TRAIN_DEVICE_ID': '/gp

Traceback (most recent call last):
  File "models/cnn_model1.py", line 79, in learn
    y = self.train()
  File "models/cnn_model1.py", line 100, in train
    logits = self.createModel(self.train_data_node, True)
  File "models/cnn_model1.py", line 377, in createModel
    hidden = tf.nn.relu(tf.matmul(reshape, self.fc1_weights) + self.fc1_biases)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/math_ops.py", line 1346, in matmul
    name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/gen_math_ops.py", line 1271, in _mat_mul
    transpose_b=transpose_b, name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/op_def_library.py", line 703, in apply_op
    op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 2312, in create_op
    set_shapes_for_outputs(ret)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 1704, in set_shap

({'conv2_depth': 64, 'dropout_rate': 0.5, 'conv1_depth': 32, 'OPTIMIZATION': 'Momentum', 'MOMENTUM_VALUE': 0.9, 'base_learning_rate': 0.01, 'filter_size': 5, 'init_stddev': 0.1, 'seed': 66478, 'BATCH_SIZE': 64, 'pooling_size': 6, 'num_epochs': 20, 'eval_frequency': 100, 'eval_batch_size': 64, 'FC1_WIDTH': 512, 'fc1_width': 512, 'momentum_value': 0.9, 'STRIDE_SIZE': 2, 'decay_rate': 0.95, 'optimization': 'Momentum', 'num_pooling': 2, 'POOLING_SIZE': 6, 'NUM_EPOCHS': 20, 'USE_LEARNING_RATE_DECAY': True, 'EVAL_FREQUENCY': 100, 'EVAL_BATCH_SIZE': 64, 'regularizer_factor': 0.0005, 'batch_size': 64, 'BASE_LEARNING_RATE': 0.01, 'padding': 'VALID', 'FILTER_SIZE': 5, 'INIT_WEIGHT_VALUE': 0.1, 'use_learning_rate_decay': True, 'CONV2_DEPTH': 64, 'DROPOUT_RATE': 0.5, 'CONV1_DEPTH': 32, 'DECAY_RATE': 0.95, 'REGULARIZER_FACTOR': 0.0005, 'INIT_STDDEV': 0.1, 'stride_size': 2, 'PADDING': 'VALID', 'SEED': 66478, 'NUM_POOLING': 2, 'init_weight_value': 0.1},)
({'VALIDATION': False, 'TRAIN_DEVICE_ID': '/gp

Traceback (most recent call last):
  File "models/cnn_model1.py", line 79, in learn
    y = self.train()
  File "models/cnn_model1.py", line 100, in train
    logits = self.createModel(self.train_data_node, True)
  File "models/cnn_model1.py", line 377, in createModel
    hidden = tf.nn.relu(tf.matmul(reshape, self.fc1_weights) + self.fc1_biases)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/math_ops.py", line 1346, in matmul
    name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/gen_math_ops.py", line 1271, in _mat_mul
    transpose_b=transpose_b, name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/op_def_library.py", line 703, in apply_op
    op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 2312, in create_op
    set_shapes_for_outputs(ret)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 1704, in set_shap

({'conv2_depth': 64, 'dropout_rate': 0.5, 'conv1_depth': 32, 'OPTIMIZATION': 'Momentum', 'MOMENTUM_VALUE': 0.9, 'base_learning_rate': 0.01, 'filter_size': 5, 'init_stddev': 0.1, 'seed': 66478, 'BATCH_SIZE': 64, 'pooling_size': 8, 'num_epochs': 20, 'eval_frequency': 100, 'eval_batch_size': 64, 'FC1_WIDTH': 512, 'fc1_width': 512, 'momentum_value': 0.9, 'STRIDE_SIZE': 2, 'decay_rate': 0.95, 'optimization': 'Momentum', 'num_pooling': 2, 'POOLING_SIZE': 8, 'NUM_EPOCHS': 20, 'USE_LEARNING_RATE_DECAY': True, 'EVAL_FREQUENCY': 100, 'EVAL_BATCH_SIZE': 64, 'regularizer_factor': 0.0005, 'batch_size': 64, 'BASE_LEARNING_RATE': 0.01, 'padding': 'VALID', 'FILTER_SIZE': 5, 'INIT_WEIGHT_VALUE': 0.1, 'use_learning_rate_decay': True, 'CONV2_DEPTH': 64, 'DROPOUT_RATE': 0.5, 'CONV1_DEPTH': 32, 'DECAY_RATE': 0.95, 'REGULARIZER_FACTOR': 0.0005, 'INIT_STDDEV': 0.1, 'stride_size': 2, 'PADDING': 'VALID', 'SEED': 66478, 'NUM_POOLING': 2, 'init_weight_value': 0.1},)
({'VALIDATION': False, 'TRAIN_DEVICE_ID': '/gp

Traceback (most recent call last):
  File "models/cnn_model1.py", line 79, in learn
    y = self.train()
  File "models/cnn_model1.py", line 100, in train
    logits = self.createModel(self.train_data_node, True)
  File "models/cnn_model1.py", line 377, in createModel
    hidden = tf.nn.relu(tf.matmul(reshape, self.fc1_weights) + self.fc1_biases)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/math_ops.py", line 1346, in matmul
    name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/gen_math_ops.py", line 1271, in _mat_mul
    transpose_b=transpose_b, name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/op_def_library.py", line 703, in apply_op
    op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 2312, in create_op
    set_shapes_for_outputs(ret)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 1704, in set_shap

({'conv2_depth': 64, 'dropout_rate': 0.5, 'conv1_depth': 32, 'OPTIMIZATION': 'Momentum', 'MOMENTUM_VALUE': 0.9, 'base_learning_rate': 0.01, 'filter_size': 5, 'init_stddev': 0.1, 'seed': 66478, 'BATCH_SIZE': 64, 'pooling_size': 10, 'num_epochs': 20, 'eval_frequency': 100, 'eval_batch_size': 64, 'FC1_WIDTH': 512, 'fc1_width': 512, 'momentum_value': 0.9, 'STRIDE_SIZE': 2, 'decay_rate': 0.95, 'optimization': 'Momentum', 'num_pooling': 2, 'POOLING_SIZE': 10, 'NUM_EPOCHS': 20, 'USE_LEARNING_RATE_DECAY': True, 'EVAL_FREQUENCY': 100, 'EVAL_BATCH_SIZE': 64, 'regularizer_factor': 0.0005, 'batch_size': 64, 'BASE_LEARNING_RATE': 0.01, 'padding': 'VALID', 'FILTER_SIZE': 5, 'INIT_WEIGHT_VALUE': 0.1, 'use_learning_rate_decay': True, 'CONV2_DEPTH': 64, 'DROPOUT_RATE': 0.5, 'CONV1_DEPTH': 32, 'DECAY_RATE': 0.95, 'REGULARIZER_FACTOR': 0.0005, 'INIT_STDDEV': 0.1, 'stride_size': 2, 'PADDING': 'VALID', 'SEED': 66478, 'NUM_POOLING': 2, 'init_weight_value': 0.1},)
({'VALIDATION': False, 'TRAIN_DEVICE_ID': '/

Traceback (most recent call last):
  File "models/cnn_model1.py", line 79, in learn
    y = self.train()
  File "models/cnn_model1.py", line 100, in train
    logits = self.createModel(self.train_data_node, True)
  File "models/cnn_model1.py", line 377, in createModel
    hidden = tf.nn.relu(tf.matmul(reshape, self.fc1_weights) + self.fc1_biases)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/math_ops.py", line 1346, in matmul
    name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/gen_math_ops.py", line 1271, in _mat_mul
    transpose_b=transpose_b, name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/op_def_library.py", line 703, in apply_op
    op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 2312, in create_op
    set_shapes_for_outputs(ret)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 1704, in set_shap

In [1]:
missing_params_list = [{'filter_size': 11, 'fc1_width': 896, 'conv1_depth': 100, 'conv2_depth': 150}, {'filter_size': 11, 'fc1_width': 256, 'conv1_depth': 100, 'conv2_depth': 350}, {'filter_size': 11, 'fc1_width': 384, 'conv1_depth': 250, 'conv2_depth': 100}, {'filter_size': 11, 'fc1_width': 896, 'conv1_depth': 350, 'conv2_depth': 100}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 450}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 100}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 200}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 300}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 400}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 50}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 500}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 150}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 250}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 450, 'conv2_depth': 350}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 100, 'conv2_depth': 350}, {'filter_size': 9, 'fc1_width': 896, 'conv1_depth': 300, 'conv2_depth': 450}, {'filter_size': 9, 'fc1_width': 768, 'conv1_depth': 300, 'conv2_depth': 200}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 400, 'conv2_depth': 450}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 400, 'conv2_depth': 200}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 400, 'conv2_depth': 300}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 400, 'conv2_depth': 400}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 400, 'conv2_depth': 500}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 400, 'conv2_depth': 150}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 400, 'conv2_depth': 250}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 400, 'conv2_depth': 350}, {'filter_size': 9, 'fc1_width': 128, 'conv1_depth': 50, 'conv2_depth': 200}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 500, 'conv2_depth': 100}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 500, 'conv2_depth': 50}, {'filter_size': 9, 'fc1_width': 256, 'conv1_depth': 500, 'conv2_depth': 150}, {'filter_size': 9, 'fc1_width': 512, 'conv1_depth': 350, 'conv2_depth': 450}, {'filter_size': 7, 'fc1_width': 128, 'conv1_depth': 450, 'conv2_depth': 50}, {'filter_size': 7, 'fc1_width': 128, 'conv1_depth': 400, 'conv2_depth': 450}, {'filter_size': 7, 'fc1_width': 128, 'conv1_depth': 400, 'conv2_depth': 400}, {'filter_size': 7, 'fc1_width': 128, 'conv1_depth': 400, 'conv2_depth': 500}, {'filter_size': 7, 'fc1_width': 896, 'conv1_depth': 350, 'conv2_depth': 400}]

In [2]:
LOG_PATH = "../log/revised-20epochs.csv"

In [6]:
def run(params, log_path):
    try:        
        generator = HPVGenerator()
        generator.setTemplate('CNN_HPV.ini')
        hpv_file = generator.generate(params, output_dir='config/')
                
        train_manager = TrainingManager(LeNet5(mnist.import_dataset()), log_path)
        train_manager.setTrainingDevices('gpu', 4)
        train_manager.setLoggingParams([ hyperparam for hyperparam in params])
        hpv = HPVManager(hpv_file, ini_dir='config/')
        return train_manager.run(hpv, 2)
           
    except:
        e = sys.exc_info()
        print("Error: " + str(e))
        traceback.print_exc()    

In [9]:
for params in missing_params_list:
    result = run(params, LOG_PATH)
    print str(params) + " -> " + str(result)

CAUTION! the logging column number or order is different
['', 'Test Error', 'Validation Error', 'Training Error', 'filter_size', 'conv1_depth', 'conv2_depth', 'fc1_width']
({'conv2_depth': 150, 'dropout_rate': 0.5, 'FC3_WIDTH': 512, 'conv1_depth': 100, 'OPTIMIZATION': 'Momentum', 'MOMENTUM_VALUE': 0.9, 'base_learning_rate': 0.01, 'filter_size': 11, 'init_stddev': 0.1, 'seed': 66478, 'BATCH_SIZE': 64, 'pooling_size': 2, 'num_epochs': 20, 'CONV3_DEPTH': 128, 'eval_frequency': 100, 'eval_batch_size': 64, 'FC1_WIDTH': 896, 'fc1_width': 896, 'fc2_width': 512, 'momentum_value': 0.9, 'STRIDE_SIZE': 2, 'decay_rate': 0.95, 'optimization': 'Momentum', 'POOLING_SIZE': 2, 'NUM_EPOCHS': 20, 'USE_LEARNING_RATE_DECAY': True, 'EVAL_FREQUENCY': 100, 'EVAL_BATCH_SIZE': 64, 'FC2_WIDTH': 512, 'regularizer_factor': 0.0005, 'batch_size': 64, 'BASE_LEARNING_RATE': 0.01, 'padding': 'SAME', 'FILTER_SIZE': 11, 'INIT_WEIGHT_VALUE': 0.1, 'use_learning_rate_decay': True, 'CONV2_DEPTH': 150, 'DROPOUT_RATE': 0.5, 'f

Traceback (most recent call last):
  File "models/cnn_model1.py", line 79, in learn
  File "models/cnn_model1.py", line 192, in train
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 382, in run
    run_metadata_ptr)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 655, in _run
    feed_dict_string, options, run_metadata)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 723, in _do_run
    target_list, options, run_metadata)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 730, in _do_call
    return fn(*args)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 712, in _run_fn
    status, run_metadata)
KeyboardInterrupt
